components/src/core/user-menu/user-menu.tsx

```diff
@@ -0,0 +1,129 @@
+    fontColor?: string;
+    // Color of icons in the bottomsheet
+    iconColor?: string;
+    // Custom menu to render in the bottomsheet
+    menu?: JSX.Element;
+    menuItems?: InfoListItemProps[];
+    // Boolean to control the bottomsheet from parent component
+    menuOpen?: boolean;
+    menuTitle?: string;
+    menuSubtitle?: string;
+    // Function to toggle menuOpen prop from parent component
+    toggleMenu?: () => void;
+    styles?: {
+        root?: ViewStyle;
+        avatar?: ViewStyle;
+    };
+};
+
+const useStyles = (
+    theme: ReactNativePaper.Theme
+): StyleSheet.NamedStyles<{
+    root: ViewStyle;
+    avatar: ViewStyle;
+}> =>
+    StyleSheet.create({
+        root: {
+            backgroundColor: theme.colors.surface,
+        },
+        avatar: {
+            width: 40,
+            height: 40,
+            borderRadius: 40,
+        },
+    });
+
+export const UserMenu: React.FC<UserMenuProps> = (props) => {
+    const theme = useTheme();
+    const {
+        avatar,
+        backgroundColor,
+        fontColor,
+        iconColor,
+        menu,
+        menuTitle,
+        menuSubtitle,
+        menuItems,
+        menuOpen = false,
+        toggleMenu = (): void => {},
+        styles = {},
+    } = props;
+    const [showBottomSheet, setShowBottomSheet] = useState(false);
+    const defaultStyles = useStyles(theme);
+
+    const openMenu = (): void => {
+        setShowBottomSheet(true);
+    };
+
+    const closeMenu = (): void => {
+        setShowBottomSheet(false);
+    };
+
+    const getMenu = useCallback((): JSX.Element => {
+        if (menu) {
+            return menu;
+        }
+
+        return (
+            <>
+                {menuTitle && (
+                    <>
+                        <InfoListItem
+                            title={menuTitle || ''}
+                            subtitle={menuSubtitle}
+                            customIcon={<View style={[defaultStyles.avatar, styles.avatar]}>{avatar}</View>}
+                            fontColor={fontColor}
+                            backgroundColor={backgroundColor}
+                        />
+                        <Divider />
+                    </>
+                )}
+                {menuItems &&
+                    menuItems.map((menuItem: InfoListItemProps, index: number) => (
+                        <InfoListItem
+                            key={index}
+                            IconClass={menuItem.IconClass}
+                            onPress={(): void => {
+                                closeMenu();
+                                if (menuItem.onPress) menuItem.onPress();
+                            }}
+                            iconColor={iconColor || menuItem.iconColor}
+                            fontColor={fontColor || menuItem.fontColor}
+                            backgroundColor={backgroundColor || menuItem.backgroundColor}
+                            {...menuItem}
```

Comment:
spread should come above the props you need to control, otherwise the user input will override everything you set (e.g. if they have an onClick, your addition of the closeMenu function won't work)